api/jsonapi.go

```diff
@@ -1613,8 +1613,8 @@ func (i *jsonAPIHandler) POSTOrderCancel(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	if !(state == pb.OrderState_PENDING && len(records) > 0) || state != pb.OrderState_PENDING || contract.BuyerOrder.Payment.Method == pb.Order_Payment_MODERATED {
-		ErrorResponse(w, http.StatusBadRequest, "order must be PENDING or partially funded and only a direct payment to cancel")
+	if !((state == pb.OrderState_PENDING || state == pb.OrderState_PROCESSING_ERROR) && len(records) > 0) || !(state == pb.OrderState_PENDING || state == pb.OrderState_PROCESSING_ERROR) || contract.BuyerOrder.Payment.Method == pb.Order_Payment_MODERATED {
```

Comment:
I think `contract.CanCancel()` might be ideal but this would require a pretty large refactor. `contract` is a protobuf object and to add a `CanCancel()` method would require extending it... which would require changing the type everywhere `pb.RicardianContract` is used in the code. 

Might be worth it later but I'd say that would be worthy of a separate PR.